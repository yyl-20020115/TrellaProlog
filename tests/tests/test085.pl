% Traversing graph paths

oneway(paris,orleans).
oneway(paris,chartres).
oneway(paris,amiens).
oneway(orleans,blois).
oneway(orleans,bourges).
oneway(blois,tours).
oneway(chartres,lemans).
oneway(lemans,angers).
oneway(lemans,tours).
oneway(angers,nantes).

path(A,B) :-
    oneway(A,B).
path(A,B) :-
    oneway(A,C),
    path(C,B).

% test cases
case(path(_,nantes)).

test :-
    case(A),
    A,
    write(A),
    write('.'),
    nl,
    fail.
test :-
    halt.

:- initialization(test).
