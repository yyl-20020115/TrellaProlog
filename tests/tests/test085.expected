path(angers,nantes).
path(paris,nantes).
path(chartres,nantes).
path(lemans,nantes).
